### PR TITLE
kubeadm: check the GV after the config migration

### PR DESCRIPTION
**Pull request by chendave**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/util/config/common_test.go [hunk 1]
```diff
@@ -451,6 +451,15 @@ func TestMigrateOldConfig(t *testing.T) {
 						t.Fatalf("migration failed to produce config kind: %s", expectedKind)
 					}
 				}
+				expectedGV := gv
+				if test.allowExperimental {
+					expectedGV = gvExperimental
+				}
+				for _, gvk := range gvks {
+					if gvk.GroupVersion().String() != expectedGV {
+						t.Fatalf("GV mismatch, expected GV: %s, got GV: %s", expectedGV, gvk.GroupVersion().String())
```
[the remark this answers is not in the document]
Comment (neolit123) [reply cmd/kubeadm/app/util/config/common_test.go:460]: /lgtm
/approve

**Comment by neolit123** [comment]
/lgtm cancel

actually see my comment above ^

File: cmd/kubeadm/app/util/config/common_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (neolit123) [reply cmd/kubeadm/app/util/config/common_test.go:460]: EDIT2: we **do** test kubelet kinds as well so the above suggestion makes sense

[the remark this answers is not in the document]
Comment (chendave) [reply cmd/kubeadm/app/util/config/common_test.go]: Done

**Comment by pacoxu** [comment]
It will fail without the fix in #119059 like below. 
```
--- FAIL: TestMigrateOldConfig (17.94s)
    --- FAIL: TestMigrateOldConfig/ClusterConfiguration_gets_migrated_from_experimental_API (2.10s)
        /Users/pacoxu/git/gopath/src/k8s.io/kubernetes/cmd/kubeadm/app/util/config/common_test.go:460: GV mismatch, expected GV: kubeadm.k8s.io/v1beta4, got GV: kubeadm.k8s.io/v1beta3
        /Users/pacoxu/git/gopath/src/k8s.io/kubernetes/cmd/kubeadm/app/util/config/common_test.go:460: GV mismatch, expected GV: kubeadm.k8s.io/v1beta4, got GV: kubeadm.k8s.io/v1beta3
FAIL
FAIL	k8s.io/kubernetes/cmd/kubeadm/app/util/config	18.767s
```

/lgtm

**Review by neolit123 — COMMENTED** [review]
/hold cancel
/approve
/lgtm
thanks

**Pull request merged**